src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#ifndef TRINITY_CHASEMOVEMENTGENERATOR_H
+#define TRINITY_CHASEMOVEMENTGENERATOR_H
+
+#include "MovementGenerator.h"
+#include "Unit.h"
+#include "Optional.h"
+
+class ChaseMovementGenerator : public MovementGenerator, AbstractFollower
+{
+    public:
+        ChaseMovementGenerator(Unit* target, Optional<float> angle) : AbstractFollower(ASSERT_NOTNULL(target)), _angle(angle) {}
+
+        void Initialize(Unit* owner) override {}
+        void Finalize(Unit* owner) override {}
+        void Reset(Unit* owner) override {}
+        bool Update(Unit* owner, uint32 diff) override;
+
+    private:
+        Optional<float> const _angle;
+
+        void startMovement(Unit* owner);
+        void pause(Milliseconds duration = 100ms)
```

Comment:
careful with naming https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Movement/MovementGenerator.h#L42

Reply:
currently implemented in WaypointMovementGenerator only, called in several places for movement slot 0